app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestActivity.java

```diff
@@ -0,0 +1,142 @@
+package io.intrepid.contest.screens.contestcreation;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.ActionBar;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import java.util.List;
+import java.util.UUID;
+
+import butterknife.BindView;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseMvpActivity;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.models.Category;
+import io.intrepid.contest.models.Contest;
+import io.intrepid.contest.screens.contestcreation.addcategoriestocontest.AddCategoriesFragment;
+import io.intrepid.contest.screens.contestcreation.describecontest.DescribeContestFragment;
+import io.intrepid.contest.screens.contestcreation.namecontest.NameContestFragment;
+import io.intrepid.contest.screens.entrysubmission.entryimage.EntryImageActivity;
+import io.intrepid.contest.screens.splash.SplashActivity;
+import io.intrepid.contest.utils.SlidingTabAdapter;
+import timber.log.Timber;
+
+public class NewContestActivity extends BaseMvpActivity<NewContestPresenter> implements NewContestContract.View, ContestActivityContract{
+
+    private static final String KEY_CREATOR_ID = "User_Id";
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+    @BindView(R.id.fragment_container)
+    ViewPager viewPager;
+    private SlidingTabAdapter tabAdapter;
+
+    public static Intent createIntent(Context context, UUID creatorId) {
+        return new Intent(context, NewContestActivity.class).putExtra(KEY_CREATOR_ID, creatorId);
+    }
+
+    @NonNull
+    @Override
+    public NewContestPresenter createPresenter(PresenterConfiguration configuration) {
+        return new NewContestPresenter(this, configuration);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.activity_viewpager_container;
+    }
+
+    public NewContestContract.Presenter getPresenter() {
+        return presenter;
+    }
+
+    public void setupViewPager(ViewPager viewPager) {
+        setToolbarTitle(R.string.new_contest);
```

Comment:
we should move this line out of this function. The function name is called "setupViewPager" but it actually sets up the view pager and sets the toolbar title. Each function should have exactly one job. We can leave setToolbarTitle inside of ShowContestSubmission Page